teuthology/report.py

```diff
@@ -262,6 +263,39 @@ def report_run(self, run_name, dead=False):
             self.log.debug("    no jobs; skipped")
         return len(jobs)
 
+    def write_new_job(self, run_name, job_info):
+        """
+        Report a new job to the results server.
+
+        :param run_name: The name of the run. The run must already exist.
+        :param job_info: The job's info dict. Must be present since this is a new job
+        """
+        if job_info is None or not isinstance(job_info, dict):
+            raise TypeError("Job info must be a dict")
+        run_uri = "{base}/runs/{name}/jobs/".format(
+            base=self.base_uri, name=run_name,
+        )
+        job_json = json.dumps(job_info)
+        headers = {'content-type': 'application/json'}
+        response = self.session.post(run_uri, data=job_json, headers=headers)
```

Comment:
like other paddles calls, I'd add a retry loop around this to work around any server side issues